webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ComplianceRepository.scala

```diff
@@ -123,37 +124,22 @@ class ComplianceJdbcRepository(
        * a run
        */
       reports.flatMap { r =>
-        r.runInfo match {
+        r.runInfo.kind match {
           // ignore case with no runs or when compliance should be kept
-          case _: KeepLastCompliance | _: NoReportInInterval | NoRunNoExpectedReport | _: ReportsDisabledInInterval |
-              _: NoUserRulesDefined =>
+          case RunAnalysisKind.KeepLastCompliance | RunAnalysisKind.NoReportInInterval | RunAnalysisKind.NoRunNoExpectedReport |
+              RunAnalysisKind.ReportsDisabledInInterval | RunAnalysisKind.NoUserRulesDefined =>
             None
-
-          case x: Pending =>
-            x.optLastRun match {
-              case None                      =>
-                None
-              case Some((runTime, expected)) =>
-                Some(RunCompliance.from(runTime, x.expirationDateTime, r))
+          case _ =>
+            (r.runInfo.lastRunDateTime, r.runInfo.expirationDateTime) match {
+              case (Some(a), Some(b)) =>
+                Some(RunCompliance.from(a, b, r))
+              case (Some(a), None)    =>
+                r.runInfo.lastRunExpiration match {
+                  case Some(b) => Some(RunCompliance.from(a, b, r))
+                  case None    => Some(RunCompliance.from(a, a.plusMinutes(10), r))
+                }
+              case _                  => None
```

Comment:
We lost the comment that explains the `10`. Is it the same value as in `ReportsDisplayer` ?